File: parser/src/main/java/dev/mvmo/sicklang/parser/ast/expression/IntegerLiteralExpressionNode.java
package dev.mvmo.sicklang.parser.ast.expression;

import dev.mvmo.sicklang.token.Token;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor(staticName = "newInstance")
public class IntegerLiteralExpressionNode implements ExpressionNode {

    private final Token token;
    private String value;

    @Override
    public String tokenLiteral() {
        return token.literal();
    }

    @Override
    public String toString() {
        return token.literal();
    }

}

File: parser/src/main/java/dev/mvmo/sicklang/parser/ast/expression/IntegerLiteralExpressionNode.kt
package dev.mvmo.sicklang.parser.ast.expression;

import dev.mvmo.sicklang.token.Token;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor(staticName = "newInstance")
public class IntegerLiteralExpressionNode implements ExpressionNode {

    private final Token token;
    private int value;

    @Override
    public String tokenLiteral() {
        return token.literal();
    }

    @Override
    public String toString() {
        return token.literal();
    }

}
